services/ai.py

```
@@ -0,0 +1,90 @@
+                    categories['personal'].append(email)
+            
+            summary = {
+                "total_emails": len(emails),
+                "categories": categories,
+                "important_emails": [email for email in emails if 'urgent' in email.get('subject', '').lower()],
+                "summary_text": f"Found {len(emails)} emails. {len(categories['work'])} work-related, {len(categories['personal'])} personal, and {len(categories['newsletters'])} newsletters."
+            }
+            
+            return summary
+        except Exception as e:
+            raise Exception(f"Error summarizing emails: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)